File: clients/typescript/src/models/MediaInfo.ts
/* tslint:disable */
/* eslint-disable */

import { mapValues, parseDate, parseDateTime, serializeDate, serializeDateTime } from '../runtime';

export interface MediaInfo {

    blobId: number;

    caption?: string | null;

    capturedAt?: Date | null;

    customerId?: string | null;

    device?: string | null;

    externalRef?: string | null;

    frameCount?: number;

    gpsLat?: number | null;

    gpsLon?: number | null;

    kind?: MediaInfoKindEnum | null;

    metadata?: { [key: string]: any; } | null;

    ocrText?: string | null;

    synthetic?: number | null;

    tags?: Array<string>;

    transcriptSegments?: number;
}

export const MediaInfoKindEnum = {
    Image: 'image',
    Video: 'video',
    Audio: 'audio',
    Pdf: 'pdf',
    Doc: 'doc',
    Note: 'note',
} as const;
export type MediaInfoKindEnum = typeof MediaInfoKindEnum[keyof typeof MediaInfoKindEnum];

export function instanceOfMediaInfo(value: object): value is MediaInfo {
    if ((!('blobId' in (value as Record<string, any>)) && !('blob_id' in (value as Record<string, any>))) || ((value as Record<string, any>)['blobId'] === undefined && (value as Record<string, any>)['blob_id'] === undefined)) return false;
    return true;
}

export function MediaInfoFromJSON(json: any): MediaInfo {
    return MediaInfoFromJSONTyped(json, false);
}

export function MediaInfoFromJSONTyped(json: any, ignoreDiscriminator: boolean): MediaInfo {
    if (json == null) {
        return json;
    }
    return {

        'blobId': json['blob_id'],
        'caption': json['caption'] === undefined ? undefined : json['caption'] === null ? null : json['caption'],
        'capturedAt': json['captured_at'] === undefined ? undefined : json['captured_at'] === null ? null : (parseDateTime(json['captured_at'])),
        'customerId': json['customer_id'] === undefined ? undefined : json['customer_id'] === null ? null : json['customer_id'],
        'device': json['device'] === undefined ? undefined : json['device'] === null ? null : json['device'],
        'externalRef': json['external_ref'] === undefined ? undefined : json['external_ref'] === null ? null : json['external_ref'],
        'frameCount': json['frame_count'] == null ? undefined : json['frame_count'],
        'gpsLat': json['gps_lat'] === undefined ? undefined : json['gps_lat'] === null ? null : json['gps_lat'],
        'gpsLon': json['gps_lon'] === undefined ? undefined : json['gps_lon'] === null ? null : json['gps_lon'],
        'kind': json['kind'] === undefined ? undefined : json['kind'] === null ? null : json['kind'],
        'metadata': json['metadata'] === undefined ? undefined : json['metadata'] === null ? null : json['metadata'],
        'ocrText': json['ocr_text'] === undefined ? undefined : json['ocr_text'] === null ? null : json['ocr_text'],
        'synthetic': json['synthetic'] === undefined ? undefined : json['synthetic'] === null ? null : json['synthetic'],
        'tags': json['tags'] == null ? undefined : json['tags'],
        'transcriptSegments': json['transcript_segments'] == null ? undefined : json['transcript_segments'],
    };
}

export function MediaInfoToJSON(json: any): MediaInfo {
    return MediaInfoToJSONTyped(json, false);
}

export function MediaInfoToJSONTyped(value?: MediaInfo | null, ignoreDiscriminator: boolean = false): any {
    if (value == null) {
        return value;
    }

    return {

        'blob_id': value['blobId'],
        'caption': value['caption'],
        'captured_at': value['capturedAt'] == null ? value['capturedAt'] : serializeDateTime(value['capturedAt']),
        'customer_id': value['customerId'],
        'device': value['device'],
        'external_ref': value['externalRef'],
        'frame_count': value['frameCount'],
        'gps_lat': value['gpsLat'],
        'gps_lon': value['gpsLon'],
        'kind': value['kind'],
        'metadata': value['metadata'],
        'ocr_text': value['ocrText'],
        'synthetic': value['synthetic'],
        'tags': value['tags'],
        'transcript_segments': value['transcriptSegments'],
    };
}

File: clients/rust/src/models/media_info.rs
use crate::models;
use serde::{Deserialize, Serialize};

#[derive(Clone, Default, Debug, PartialEq, Serialize, Deserialize)]
pub struct MediaInfo {
    #[serde(rename = "blob_id")]
    pub blob_id: i32,
    #[serde(rename = "caption", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub caption: Option<Option<String>>,
    #[serde(rename = "captured_at", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub captured_at: Option<Option<chrono::DateTime<chrono::FixedOffset>>>,
    #[serde(rename = "customer_id", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub customer_id: Option<Option<String>>,
    #[serde(rename = "device", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub device: Option<Option<String>>,
    #[serde(rename = "external_ref", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub external_ref: Option<Option<String>>,
    #[serde(rename = "frame_count", skip_serializing_if = "Option::is_none")]
    pub frame_count: Option<i32>,
    #[serde(rename = "gps_lat", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub gps_lat: Option<Option<f64>>,
    #[serde(rename = "gps_lon", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub gps_lon: Option<Option<f64>>,
    #[serde(rename = "kind", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub kind: Option<Option<Kind>>,
    #[serde(rename = "metadata", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub metadata: Option<Option<std::collections::HashMap<String, serde_json::Value>>>,
    #[serde(rename = "ocr_text", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub ocr_text: Option<Option<String>>,
    #[serde(rename = "synthetic", default, with = "::serde_with::rust::double_option", skip_serializing_if = "Option::is_none")]
    pub synthetic: Option<Option<f64>>,
    #[serde(rename = "tags", skip_serializing_if = "Option::is_none")]
    pub tags: Option<Vec<String>>,
    #[serde(rename = "transcript_segments", skip_serializing_if = "Option::is_none")]
    pub transcript_segments: Option<i32>,
}

impl MediaInfo {
    pub fn new(blob_id: i32) -> MediaInfo {
        MediaInfo {
            blob_id,
            caption: None,
            captured_at: None,
            customer_id: None,
            device: None,
            external_ref: None,
            frame_count: None,
            gps_lat: None,
            gps_lon: None,
            kind: None,
            metadata: None,
            ocr_text: None,
            synthetic: None,
            tags: None,
            transcript_segments: None,
        }
    }
}
#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Serialize, Deserialize)]
pub enum Kind {
    #[serde(rename = "image")]
    Image,
    #[serde(rename = "video")]
    Video,
    #[serde(rename = "audio")]
    Audio,
    #[serde(rename = "pdf")]
    Pdf,
    #[serde(rename = "doc")]
    Doc,
    #[serde(rename = "note")]
    Note,
}

impl Default for Kind {
    fn default() -> Kind {
        Self::Image
    }
}

